external-crates/move/crates/move-compiler/src/typing/core.rs

```diff
@@ -187,31 +161,44 @@ impl<'env> Context<'env> {
             cur.count -= 1;
             return;
         }
-        let UseFunsScope { unused, .. } = self.use_funs.pop().unwrap();
-        let cur_depth = self.use_funs.len();
-        let mut new_unused = BTreeMap::new();
-        for ((tn, method), (loc, kind, depth)) in unused {
-            if depth != cur_depth {
-                // the unused was added in a scope above
-                assert!(depth < cur_depth);
-                new_unused.insert((tn, method), (loc, kind, depth));
-                continue;
+        let UseFunsScope { use_funs, .. } = self.use_funs.pop().unwrap();
+        for (tn, methods) in use_funs {
+            let unused = methods.iter().filter(|(_, _, uf)| !uf.used);
+            for (_, method, use_fun) in unused {
+                let N::UseFun {
+                    loc,
+                    kind,
+                    attributes: _,
+                    is_public: _,
+                    target_function: _,
+                    used: _,
+                } = use_fun;
+                let msg = match kind {
+                    UseFunKind::Explicit => {
+                        format!("Unused 'use fun' of '{tn}.{method}'. Consider removing it")
+                    }
+                    UseFunKind::UseAlias => {
+                        format!("Unused 'use' of alias '{method}'. Consider removing it")
+                    }
+                    UseFunKind::FunctionDeclaration => {
+                        panic!("ICE function declaration use funs should never be added to use fun")
+                    }
+                };
+                self.env.add_diag(diag!(UnusedItem::Alias, (*loc, msg)))
             }
-            let msg = match kind {
-                UseFunKind::Explicit => {
-                    format!("Unused 'use fun' of '{tn}.{method}'. Consider removing it")
-                }
-                UseFunKind::UseAlias { used } => {
-                    assert!(!used);
-                    format!("Unused 'use' of alias '{method}'. Consider removing it")
-                }
-                UseFunKind::FunctionDeclaration => {
-                    panic!("ICE function declaration use funs should never be added to use fun")
-                }
-            };
-            self.env.add_diag(diag!(UnusedItem::Alias, (loc, msg)))
         }
-        self.use_funs.last_mut().unwrap().unused = new_unused;
+    }
+
+    pub fn find_method_and_mark_used(
+        &mut self,
+        tn: &TypeName,
+        method: Name,
+    ) -> Option<(ModuleIdent, FunctionName)> {
+        self.use_funs.iter_mut().rev().find_map(|scope| {
```

Comment:
`find_map`? I guess I have to go rewrite code to use that, having not known its existence before...